Update test_tree.py

AliAmouz opened the pull request:
Instead of having a test method for every test case, I thought about using parameterized tests to avoid duplications and make it easier to add new test .

AliAmouz committed:
Update test_tree.py

parameterized tests in test tree

yassineibr commented:
can you check the ci and try to fix the issues that are making the ci fail

p3nal reviewed (changes requested):
All good but need to install parameterized module in the test pipeline

yassineibr commented:
I also suggest to not use the parameterized module, because we don't have a requirement file ( as this app does not depend on external libraries). We can have just a for loop that do the same thing as the entire module.

AliAmouz committed:
Update test_tree.py __2__:)

a simple for loop is a very good idea (  generated AI code goes crazy).I really appreciate the aim to avoid external libraries and importations . And I hope the issue is resolved.
